sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-active-supplier-found-processor.ts

```diff
@@ -0,0 +1,103 @@
+export class CASActiveSupplierFoundProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Update student supplier based on the supplier information found on CAS.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param _auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    _auth: CASAuthDetails,
+    summary: ProcessSummary,
+  ): Promise<ProcessorResult> {
+    if (evaluationResult.status !== CASEvaluationStatus.ActiveSupplierFound) {
+      throw new Error("Incorrect CAS evaluation result processor selected.");
+    }
+    summary.info("Active CAS supplier found.");
+    try {
+      const supplierToUpdate = evaluationResult.activeSupplier;
+      let supplierAddressToUpdate: SupplierAddress = null;
+      if (evaluationResult.matchedAddress) {
+        // If no matched address sites was found, the address information will be
+        // persisted as null and the supplier isValid will be set to false for now
+        // till the CAS API call for "create site" is implemented.
+        // TODO: implement site creation on CAS.
+        const address = evaluationResult.matchedAddress;
+        supplierAddressToUpdate = {
+          supplierSiteCode: address.suppliersitecode,
+          addressLine1: address.addressline1,
+          addressLine2: address.addressline2,
+          city: address.city,
+          provinceState: address.province,
+          country: address.country,
+          postalCode: address.postalcode,
+          status: address.status,
+          siteProtected: address.siteprotected,
+          lastUpdated: new Date(address.lastupdated),
+        };
+      }
+      const now = new Date();
+      const systemUser = this.systemUsersService.systemUser;
+      const updateResult = await this.casSupplierRepo.update(
+        {
+          id: casSupplier.id,
+        },
+        {
+          supplierNumber: supplierToUpdate.suppliernumber,
+          supplierName: supplierToUpdate.suppliername,
+          status: supplierToUpdate.status,
+          supplierProtected: supplierToUpdate.supplierprotected === "Y",
+          lastUpdated: new Date(supplierToUpdate.lastupdated),
+          supplierAddress: supplierAddressToUpdate,
+          supplierStatus: SupplierStatus.Verified,
+          supplierStatusUpdatedOn: now,
+          isValid: !!supplierAddressToUpdate,
+          updatedAt: now,
+          modifier: systemUser,
+        },
+      );
+      if (updateResult.affected) {
+        summary.info("Updated CAS supplier for the student.");
```

Comment:
Will it be good to identify the student with student id?

Reply:
The supplier id (which) will be related to a student is logged once to the summary to avoid repeating it.
Since the summary is grouped I do not see a need to log it again. It may need a "correlation id" for the logs, but if the id is required here it would be required in all other logs. Does it make sense?

![image](https://github.com/user-attachments/assets/f2fc6352-b293-4518-b8fd-db498bfb8639)